src/libostree/ostree-repo-libarchive.c

```diff
@@ -247,6 +253,18 @@ aic_get_final_path (OstreeRepoArchiveImportContext *ctx,
                     const char  *path,
                     GError     **error)
 {
+  if (ctx->opts->translate_pathname)
+    {
+      struct stat stbuf;
+      path = path_relative (path, error);
+      read_archive_entry_stat (ctx->entry, &stbuf);
+      GFileType ftype = ot_gfiletype_from_stbuf (&stbuf);
+      char *ret = ctx->opts->translate_pathname (ctx->repo, ftype, path,
+                                                 ctx->opts->translate_pathname_user_data);
+      if (ret)
+        return ret;
+      /* Fall through */
+    }
```

Comment:
I'd say we add an `else` here. I.e. if `translate_pathname` is active, `use_ostree_convention` has no effect. Otherwise, it's odd thinking that there's a difference between returning `NULL` and returning the exact same string.